sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -866,7 +866,14 @@ case class CacheTableAsSelect(
     plan: LogicalPlan,
     originalText: String,
     isLazy: Boolean,
-    options: Map[String, String]) extends LeafCommand
+    options: Map[String, String],
+    isPlanAnalyzed: Boolean = false) extends Command with TransformationAfterAnalysis {
```

Comment:
`LeafCommand` refactor was just introduced. Maybe we should introduce `ModifableLeafCommand`?